src/standard_library/array_methods.md

```diff
@@ -0,0 +1,150 @@
+
+# Array
+
+For convenience, the STD provides some ready-to-use, common methods for arrays[^migrationNote]:
+
+## len
+
+Returns the length of an array
+
+```rust
+fn len<T, N>(_array: [T; N]) -> comptime Field
+```
+
+example
+
+```rust
+fn main() {
+    let array = [42, 42]
+    constrain arr.len() == 2;
+}
+```
+
+## sort
+
+Returns a new sorted array. Original array remains untouched. Notice that this function will only work for arrays of fields or integers, not for any arbitrary type. This is because the sorting logic it uses internally is optimized specifically for these values. If you need a sort function to sort any type, you should use the below function `sort_via`.
```

Comment:
```suggestion
Returns a new sorted array. The original array remains untouched. Notice that this function will only work for arrays of fields or integers, not for any arbitrary type. This is because the sorting logic it uses internally is optimized specifically for these values. If you need a sort function to sort any type, you should use the function `sort_via` described below.
```
Grammar edit